lib/absinthe/schema/notation/writer.ex

```diff
@@ -29,10 +31,22 @@ defmodule Absinthe.Schema.Notation.Writer do
         def __absinthe_errors__, do: unquote(errors)
         def __absinthe_interface_implementors__, do: unquote(implementors)
         def __absinthe_exports__, do: unquote(exports)
-      end
+      end,
+      custom_default_resolve(default_resolve_func)
     ]
   end
 
+  defp custom_default_resolve(nil) do
+    quote do
+      def __absinthe_custom_default_resolve__, do: nil
```

Comment:
My first go at this had a `__absinthe_default_resolve__` that would return the built-in default resolver unless it had been overridden. The fact we can skip argument building for the built-in default resolver meant just using this for the custom resolver made more sense, performance-wise.